cmake/install-rules.cmake

```diff
@@ -1,72 +1,50 @@
-# Installation {
-
-# Include module for standard, cross-platform install paths
-include(GNUInstallDirs)
-
-# Layout. This works for all platforms:
-#   * ${CMAKE_INSTALL_LIBDIR}/cmake/<PROJECT-NAME>
-#   * ${CMAKE_INSTALL_INCLUDEDIR}/fplus/
-set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
-set(include_install_dir "${CMAKE_INSTALL_INCLUDEDIR}")
-
-set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
-
-# Configuration
-set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
-set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
-set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
-set(namespace "${PROJECT_NAME}::")
-
-# Include module with fuction 'write_basic_package_version_file'
 include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
 
-# Configure '<PROJECT-NAME>ConfigVersion.cmake'
-# Use:
-#   * PROJECT_VERSION
-write_basic_package_version_file(
-    "${version_config}" COMPATIBILITY SameMajorVersion ARCH_INDEPENDENT
-)
-
-# Configure '<PROJECT-NAME>Config.cmake'
-# Use variables:
-#   * TARGETS_EXPORT_NAME
-configure_file("cmake/Config.cmake.in" "${project_config}" @ONLY)
-
-# Targets:
-#   * header location after install: ${CMAKE_INSTALL_INCLUDEDIR}/fplus/fplus.hpp
-#   * headers can be included by C++ code `#include <fplus/fplus.hpp>`
+# Create an export set for the CMake package
 install(
     TARGETS fplus
-    EXPORT "${TARGETS_EXPORT_NAME}"
-    INCLUDES DESTINATION "${include_install_dir}"
+    EXPORT FunctionalPlusTargets
+    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
 )
 
 # Dev component name, useful if the client embeds the library into their build
 # tree and want to install only their own files
-set(fplus_component "${PROJECT_NAME}_Development")
+set(fplus_component FunctionalPlus_Development)
 
-# Headers:
-#   * include/fplus/fplus.hpp -> ${CMAKE_INSTALL_INCLUDEDIR}/fplus/fplus.hpp
+# Install the include/fplus directory to a location that was added to the
+# export set and the include directories of its target above
 install(
-    DIRECTORY "include/fplus" # no trailing slash
-    DESTINATION "${include_install_dir}"
+    DIRECTORY "${PROJECT_SOURCE_DIR}/include/fplus" # no trailing slash
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     COMPONENT "${fplus_component}"
 )
 
-# Config
-#   * ${CMAKE_INSTALL_LIBDIR}/cmake/FunctionalPlus/FunctionalPlusConfig.cmake
-#   * ${CMAKE_INSTALL_LIBDIR}/cmake/FunctionalPlus/FunctionalPlusConfigVersion.cmake
+# Use the PROJECT_VERSION variable to generate a basic config version file in
+# the PROJECT_BINARY_DIR directory
+write_basic_package_version_file(
+    FunctionalPlusConfigVersion.cmake
+    COMPATIBILITY SameMajorVersion
```

Comment:
Maybe we can even use the minor here? @Dobiasd all versions 0.2.x are more or less compatible?

Reply:
Is there a need to steer away from semver?

Reply:
> @Dobiasd all versions 0.2.x are more or less compatible?

Yeah, they should be. I just went through the changelogs, and I think it was all:
- bug fixes
- new functionality
- performance improvements
- docs
- build system

I can't guarantee 100% backward compatibility, but there were no things breaking it intentionally. So it'd bluntly answer with "Yes.". :slightly_smiling_face:

Reply:
For the record, here is the documentation for [write_basic_package_version_file](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file) that explains what the `COMPATIBILITY` argument means.

Reply:
> Is there a need to steer away from semver?

No, just an idea. It shouldn't block the PR btw.

Reply:
Shall we resolve this conversation and merge the PR?

Reply:
Yes, please.